Feature/san 5100 wallets trigger

GarageInc opened the pull request:
1. Refactored styles for trigger form(block of errors, checkboxes and etc)
2. Restored charts, fixed by last changes in charts
3. I did't find a quick css-solution for dynamic validation errors and form fields with/without labels. And I add nbsp labels, will be removed in future.
4. Editing trigger in a page as modal, using Dialog
![image](https://user-images.githubusercontent.com/14061779/59751658-9afda600-9289-11e9-8702-43bb3603659f.png)

![image](https://user-images.githubusercontent.com/14061779/59671267-1138d480-91c6-11e9-9cd3-cb8e83ae3f39.png)
![image](https://user-images.githubusercontent.com/14061779/59671292-18f87900-91c6-11e9-80da-d2a65e58a2a2.png)
![image](https://user-images.githubusercontent.com/14061779/59671324-2281e100-91c6-11e9-91e1-acc3447f618d.png)

Partysun commented on src/ducks/SANCharts/ChartMetrics.js:
```diff
@@ -63,5 +67,11 @@ class ChartMetrics extends Component {
+  skip: ({ showOnlyDefault }) => {
+    return showOnlyDefault
+  },
+  options: data => {
```
({ slug }) ?

GarageInc replied:
Will change. Remaining from debugging.

DmitriVanGuard commented on src/ducks/Signals/signalFormManager/signalCrudForm/signal/TriggerForm.module.scss:
```diff
@@ -65,14 +69,18 @@ a.connectLink {
 .notifyBlock {
   display: flex;
+
+  div:first-child {
```
Could it be rewritten without using tag selector and without nesting

DmitriVanGuard commented on src/ducks/Signals/signalModal/SignalMasterModalForm.js:
```diff
@@ -1,46 +1,82 @@
 import { connect } from 'react-redux'
-import { Modal, Button, Icon } from '@santiment-network/ui'
+import { Button, Icon } from '@santiment-network/ui'
+import Dialog from '@santiment-network/ui/Dialog'
```
Use either default import (better, because of the tree shaking) or import multiple exports

DmitriVanGuard commented on src/ducks/Signals/signalModal/SignalMasterModalForm.js:
```diff
@@ -1,46 +1,82 @@
+
+const signalModalTrigger = (isLoggedIn, label, canShow) =>
+  !canShow ? (
+    <div />
```
Just return `null`/`undefined`

GarageInc replied:
It is a trigger for dialog. Dialog crashes if trigger return null/undefined. Need to show dialog without a button/trigger for dialog.
![image](https://user-images.githubusercontent.com/14061779/59742068-6ed82a00-9275-11e9-996a-ba5c0b979b91.png)

DmitriVanGuard replied:
Then don't render `Dialog` itself in such case

DmitriVanGuard commented on src/ducks/Signals/utils/utils.js:
```diff
@@ -566,3 +594,12 @@ export const validateTriggerForm = values => {
 }
+
+export const couldShowChart = metric => {
+  const possibleForChart = [
```
Why the `possibleForChart` should be initialised on every function call? Could it be moved outside the function?

GarageInc commented on src/ducks/Signals/signalFormManager/SignalMaster.js:
```diff
@@ -4,58 +4,67 @@ import { graphql } from 'react-apollo'
   mapGQLTriggerToProps
 } from '../utils/utils'
 import styles from './signalCrudForm/signal/TriggerForm.module.scss'
+import { SIGNAL_ROUTES } from '../common/constants'
```
Ok. The first time I see someone watching the order of imports :D

GarageInc commented on src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormMetricValues.js:
```diff
@@ -36,10 +48,11 @@ export const TriggerFormMetricValues = ({
           'absoluteBorderRight'
         ) && (
         <div className={styles.Field}>
+          <label>&nbsp;</label>
```
I did't find a quick css-solution for dynamic validation errors and form fields with/without labels.  And I add nbsp labels, will be removed in future. It works correct with such labels in different modes.

GarageInc commented:
I added &nbsp to labels, @DmitriVanGuard , because I did't find a quick css-solution for dynamic validation errors(texts) and form fields with/without labels in different modes. And I added labels with  &nbsp, they will be removed in future when I find a right solution.

Partysun commented on src/ducks/Signals/signalFormManager/SignalMaster.js:
```diff
@@ -89,37 +97,39 @@ export class SignalMaster extends React.PureComponent {
+        {step === STEPS.SETTINGS && (
+          <TriggersForm
+            onClose={() => {
+              close()
```
Just close

Partysun commented on src/components/SignalCard/controls/SignalControls.js:
```diff
@@ -0,0 +1,62 @@
+  let iconType
+  let iconStyle
+
+  switch (type) {
```
You can make a separate function for this and reduce let usage.

DmitriVanGuard commented on src/ducks/Signals/signalFormManager/signalCrudForm/signal/TriggerForm.js:
```diff
@@ -131,6 +141,25 @@ export const TriggerForm = ({
+                  false
+                )
+
+                !!current.values.target &&
```
If the value is `truthy`, then `!!` will just convert it to `true` and that's not necessary

DmitriVanGuard commented on src/ducks/Signals/signalFormManager/signalCrudForm/signal/TriggerForm.module.scss:
```diff
@@ -31,25 +39,22 @@
+  @include text('body-3');
+
+  position: relative;
+  right: 10px;
```
Could it be rewritten with `margin` or `padding`?

DmitriVanGuard commented on src/ducks/Signals/utils/utils.js:
```diff
@@ -114,25 +123,26 @@ const getTriggerOperation = ({
   const mapped = {}
 
-  switch (type.type) {
+  const value = type.value
```
`const {value} = type`

DmitriVanGuard commented on src/ducks/Signals/utils/utils.js:
```diff
@@ -212,6 +222,10 @@ const getTriggerToFormThreshold = ({ threshold, operation }) => {
   return newThreshold
 }
 
+function capitalizeFirstLetter (string) {
```
@Partysun Isn't this kind of function already declared in `utils`?

DmitriVanGuard replied:
[It is](https://github.com/santiment/app/blob/8adc341f6bd58067e5fb462cd2d35678dc25e2f7/src/utils/utils.js#L233)

Partysun replied:
@GarageInc ☝️

DmitriVanGuard commented on src/pages/SonarFeed/SonarFeedActivityPage.js:
```diff
@@ -38,8 +38,8 @@ const SonarFeedActivityPage = ({ activities, isLoading, isError, isEmpty }) => {
-      {activities.map(activity => (
-        <Fragment key={activity.triggeredAt}>
+      {activities.map((activity, index) => (
+        <Fragment key={index}>
```
@Partysun @GarageInc Could `activities` be sorted? If they are, then using `index` as key is a bad practice

Partysun replied:
@DmitriVanGuard Should be sorted by date. First item is the last activity.
But it can be fixed in a separate PR. Cause full view of activities must be made with our design, not as we do now.
